File: src/modifiers/presentationCornerRadius.ts
import { createModifier } from './createModifier';

/// SwiftUI `.presentationCornerRadius(_:)`. Requests the corner radius of
/// the presentation (sheet) it is applied to. Pass `undefined` (or omit)
/// to restore the system default.
///
/// NOTE: `presentationCornerRadius` is **not available on watchOS**
/// (iOS 16.4+ / macOS 13.3+ only). On watchOS this modifier is a
/// documented no-op — accepted for cross-platform JS parity, applies
/// nothing.
export function presentationCornerRadius(radius?: number) {
  return createModifier('presentationCornerRadius', { radius });
}

File: apple/Sources/ReactNativeWatchOS/Modifiers/PresentationCornerRadius.swift
import SwiftUI

/// SwiftUI `.presentationCornerRadius(_:)`.
///
/// LIMITATION: `presentationCornerRadius` is **iOS 16.4+ / macOS 13.3+
/// only — it is not available on watchOS** (the symbol cannot be
/// referenced here). watchOS sheets use the system presentation shape and
/// expose no corner-radius override, so this modifier is registered as a
/// no-op: it accepts the `radius` param for cross-platform JS parity and
/// returns the view unchanged.
enum RNWPresentationCornerRadiusModifier {
    @MainActor
    static func register(into r: RNWModifierRegistry) {
        r.register("presentationCornerRadius") { view, _, _ in
            // No-op on watchOS — `presentationCornerRadius` is unavailable here.
            view
        }
    }
}
